session/src/main/java/org/apache/iotdb/session/SessionUtils.java

```diff
@@ -58,42 +59,95 @@ public static ByteBuffer getValueBuffer(Tablet tablet) {
 
   private static void getValueBufferOfDataType(
       TSDataType dataType, Tablet tablet, int i, ByteBuffer valueBuffer) {
+    int lastPos = 0, nextPos;
+    BitSet curBitset = tablet.BitSets[i];
+
     switch (dataType) {
       case INT32:
         int[] intValues = (int[]) tablet.values[i];
-        for (int index = 0; index < tablet.rowSize; index++) {
-          valueBuffer.putInt(intValues[index]);
+        while (true) {
+          nextPos = curBitset.nextClearBit(lastPos);
+          for (int index = lastPos; index < nextPos; index++) {
+            valueBuffer.putInt(intValues[index]);
+          }
+          if (nextPos == tablet.rowSize) {
+            break;
+          }
+          valueBuffer.putInt(-1);
+          lastPos = nextPos + 1;
```

Comment:
Is there any way to put the bitSet into valueBuffer? The server needs to get the bitSet to know which value is valid....